src/plugins/controls/public/options_list/components/options_list_popover_suggestions.tsx

```diff
@@ -49,130 +43,137 @@ export const OptionsListPopoverSuggestions = ({
+    showOnlySelected,
+    selectedOptionsSet,
+    invalidSelectionsSet,
+    existsSelectableOption,
+  ]);
+
+  const [suggestionsTest, setSuggestionsTest] = useState<EuiSelectableOption[]>([]); // will be set in following useEffect
+  useEffect(() => {
+    /* This useEffect makes suggestionsSelectableOptions responsive to search, show only selected, and clear selections */
+    setSuggestionsTest(suggestionsSelectableOptions);
+  }, [suggestionsSelectableOptions]);
 
   return (
     <>
-      {!hideExists && !(showOnlySelected && !existsSelected) && (
-        <EuiFilterSelectItem
-          data-test-subj={`optionsList-control-selection-exists`}
-          checked={existsSelected ? 'on' : undefined}
-          key={'exists-option'}
-          onClick={() => {
+      <EuiSelectable
+        isLoading={isLoading}
+        options={suggestionsTest}
+        aria-label={OptionsListStrings.popover.getSuggestionsAriaLabel(fieldName)}
+        listProps={{ onFocusBadge: false }} // isVirtualized: false
+        emptyMessage={
+          <span
+            className="euiFilterSelect__note"
+            data-test-subj={`optionsList-control-${
+              showOnlySelected ? 'selectionsEmptyMessage' : 'noSelectionsMessage'
+            }`}
+          >
+            <span className="euiFilterSelect__noteContent">
+              <EuiIcon type="minusInCircle" />
+              <EuiSpacer size="xs" />
+              {showOnlySelected
+                ? OptionsListStrings.popover.getSelectionsEmptyMessage()
+                : OptionsListStrings.popover.getEmptyMessage()}
+            </span>
+          </span>
+        }
+        onChange={(newSuggestions, _, changedOption) => {
+          setSuggestionsTest(newSuggestions);
+
+          const key = changedOption.key ?? changedOption.label;
+          if (key === 'exists-option') {
             dispatch(selectExists(!Boolean(existsSelected)));
-          }}
-          className="optionsList__existsFilter"
-        >
-          {OptionsListStrings.controlAndPopover.getExists()}
-        </EuiFilterSelectItem>
-      )}
-      {suggestions?.map((key: string) => (
-        <EuiFilterSelectItem
-          data-test-subj={`optionsList-control-selection-${key}`}
-          checked={selectedOptionsSet?.has(key) ? 'on' : undefined}
-          key={key}
-          onClick={() => {
-            if (showOnlySelected) {
-              dispatch(deselectOption(key));
-              return;
-            }
-            if (singleSelect) {
-              dispatch(replaceSelection(key));
-              return;
-            }
-            if (selectedOptionsSet.has(key)) {
-              dispatch(deselectOption(key));
-              return;
-            }
+          } else if (showOnlySelected) {
+            dispatch(deselectOption(key));
+          } else if (selectedOptionsSet.has(key)) {
+            dispatch(deselectOption(key));
+          } else if (singleSelect) {
+            dispatch(replaceSelection(key));
```

Comment:
Note that the `singleSelect` check was moved down one level because the previous order of checks was causing a bug where, when `singleSelect` was on, selections could not be undone unless `showOnlySelected` was `true`.